python/orca/src/bigdl/orca/data/shard.py

```diff
@@ -170,7 +173,8 @@ def transform(iter, func, *args):
                 yield func(x, *args)
 
         transformed_shard = SparkXShards(self.rdd.mapPartitions(lambda iter:
-                                                                transform(iter, func, *args)))
+                                                                transform(iter, func, *args)),
+                                         class_name = self.class_name)
```

Comment:
this is not correct - the transformed class maybe different from the original class